initrd/directory.go

```diff
@@ -54,10 +55,11 @@ func (initrd *directory) Build(ctx context.Context) (string, error) {
 	if initrd.opts.output == "" {
 		fi, err := os.CreateTemp("", "")
 		if err != nil {
-			return "", fmt.Errorf("could not make temporary directory: %w", err)
+			return "", fmt.Errorf("could not make temporary file: %w", err)
 		}
 
 		initrd.opts.output = fi.Name()
+		defer fi.Close()
```

Comment:
Nice catch ;)

Reply:
Wait, this should actually not be in a `defer`. We are re-opening that file immediately below, so we should be closing it right away.

Reply:
That's how I had it initially, but wanted to have less lines in the PRs, will change ;)

Reply:
Changed